app/services/AtsService.scala

```diff
@@ -94,19 +95,24 @@ class AtsService @Inject()(
       case individualUtr: SaUtr => middleConnector.connectToAts(individualUtr, taxYear)
     }
 
-    gotData flatMap { data =>
-      data.errors match {
-        case None =>
+    EitherT {
+      gotData flatMap {
+        case AtsSuccessResponseWithPayload(data: AtsData) if hasNoAts(data) => Future.successful(Left(NOT_FOUND))
+        case AtsSuccessResponseWithPayload(data: AtsData) if data.errors.nonEmpty =>
+          Future.successful(Left(INTERNAL_SERVER_ERROR))
+        case AtsSuccessResponseWithPayload(data: AtsData) =>
           sendAuditEvent(account, data)
-          storeAtsData(data)
-        case Some(IncomingAtsError("NoAtsError")) =>
-          storeAtsData(data)
-        case Some(_) =>
-          Future.successful(data)
+          storeAtsData(data) map (Right(_))
```

Comment:
Should map over `sendAuditEvent` this would ensure that we are always completing the audit (even if `storeAtsData` throws) and also helps us with race conditions in our tests that are expecting the audit Future to complete by the time the test completes. 

```
          for {
            result <- storeAtsData(data) map (Right(_))
            _ <- sendAuditEvent(account, data)
          } yield result
```